srtcore/api.cpp

```diff
@@ -233,39 +243,112 @@ string srt::CUDTUnited::CONID(SRTSOCKET sock)
     return os.str();
 }
 
-int srt::CUDTUnited::startup()
+bool srt::CUDTUnited::startGarbageCollector()
 {
-    ScopedLock gcinit(m_InitLock);
+    bool ret = false;
+    if (!m_bGCStatus)
+    {
+        m_bGCStatus = true;
+        m_bClosing = false;
+        ret = StartThread(m_GCThread, garbageCollect, this, "SRT:GC");
+        m_bGCStatus = ret;
```

Comment:
I think it would be clearer if you call StartThread, check its return value, and then modify `m_bClosing` and `m_bGCStatus` only in case when this function returned `true`.